### ISSUE TITLE
1.2.0 deletes the config directory contents instead of moving them

### ISSUE THREAD

**Issue by prurigro** [00-Evan/shattered-pixel-dungeon#939]
Hi! I should start by saying how impressed I am by where you've taken the game since I'd previously played- I recently got a pinephone pro and have been doing runs (with the desktop version!) after not playing for a year or so, and all the improvements you've made just blew me away! You got me excited enough about the project again that I finally got around to subscribing to your patreon :)

So here's the issue I ran into: I just upgraded the archlinux package of shattered I maintain (shattered-pixel-dungeon-git in the AUR) to 1.2.0, and when I ran it, the contents of ~/.shatteredpixel/ were deleted and the game opened like I was running it for the first time. I haven't looked at the code, but after a small amount of investigation I suspect it's supposed to move ~/.shatteredpixel/* to ~/.local/share/.shatteredpixel/, but for some reason simply deletes the contents of ~/.shatteredpixel/ and creates a new config @ ~/.local/share/.shatteredpixel/. Thankfully I make backups and was able to test moving the contents prior to running, and it loaded my old config correctly, but that migration step is likely going to delete peoples saves the way it currently is.

For now, I'll keep a note in my package warning people upgrading from 1.1.2 to backup and move their config in advance.

On a side note, I'd recommend making the config location on linux ~/.local/share/shatteredpixel instead of ~/local/share/.shatteredpixel since dot files/folders aren't really conventional in there.

Thanks for all the hard work and passion you pour into this project!

**Comment by 00-Evan** [comment]
Here's the relevant snipped of code:

```
String XDGHome = System.getenv().get("XDG_DATA_HOME");
if (XDGHome == null) XDGHome = ".local/share/";
basePath = XDGHome + ".shatteredpixel/shattered-pixel-dungeon/";

//copy over files from old linux save DIR, pre-1.2.0
FileHandle oldBase = new Lwjgl3FileHandle(".shatteredpixel/shattered-pixel-dungeon/", Files.FileType.External);
FileHandle newBase = new Lwjgl3FileHandle(XDGHome + ".shatteredpixel/shattered-pixel-dungeon/", Files.FileType.External);
if (oldBase.exists()){
        if (newBase.exists()){
	        oldBase.deleteDirectory();
        } else {
	        oldBase.moveTo(newBase);
        }
}
```

In essence, if the old config location exists, then the game moves it to the new location, if both exist it assumes the files at the new location are newer and just deletes the old data.

**Comment by prurigro** [comment]
The package I maintain is just `./gradlew desktop:release` with a small tweak to the gradle.properties changing `org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8` to `org.gradle.jvmargs=-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8` - so in theory this should be what everyone's going to get.

I just tested again by ensuring ~/.local/share/.shatteredpixel didn't exist and placed my save folder @ ~/.shatteredpixel before starting the game. The result is that ~/.shatteredpixel still exists but is now empty, and ~/.local/share/.shatteredpixel is created but has a fresh save in it.

Reading the snippet you pasted, I think (after a small amount of testing) I figured out what the issue is. If I do the same as above except first create ~/.local/share/.shatteredpixel, the save is migrated successfully and ~/.shatteredpixel still exists. I think it's trying to move ~/.shatteredpixel/shattered-pixel-dungeon to ~/.local/share/.shatteredpixel/, but .shatteredpixel doesn't exist in ~/.local/share yet, so it can't move the subfolder. That would also explain why ~/.shatteredpixel continues to exist.

If you instead moved ~/.shatteredpixel to ~/.local/share/, I suspect it would both migrate the config successfully and not leave ~/.shatteredpixel empty in the home directory. I might also suggest checking to see if ~/.local/share exists and creating it if it doesn't for good measure.
